### PR TITLE
Fix #995: off by one on validity short-circuit

### LINKED ISSUE #995: Change in behavior of LineString::simplify from 0.23.1 to 0.24.0
The results of calling `simplify` on a linestring has changed from version 0.23.1 to 0.24.0 - at least sometimes, it appears to be doing nothing at all.

I have a reproduction here: https://github.com/rjmac/simplify - it just does

```rust
let unsimplified = line_string![
    (x: 0.0, y: 0.0),
    (x: 5.0, y: 4.0),
    (x: 11.0, y: 5.5),
    (x: 17.3, y: 3.2),
    (x: 27.8, y: 0.1)
];
let simplified = unsimplified.simplify(&30.0);
println!("Before: {unsimplified:?}");
println!("After: {simplified:?}");
 ```
With geo-23 it prints
```
Before: LineString([Coord { x: 0.0, y: 0.0 }, Coord { x: 5.0, y: 4.0 }, Coord { x: 11.0, y: 5.5 }, Coord { x: 17.3, y: 3.2 }, Coord { x: 27.8, y: 0.1 }])
After: LineString([Coord { x: 0.0, y: 0.0 }, Coord { x: 27.8, y: 0.1 }])
```
and with 24,
```
Before: LineString([Coord { x: 0.0, y: 0.0 }, Coord { x: 5.0, y: 4.0 }, Coord { x: 11.0, y: 5.5 }, Coord { x: 17.3, y: 3.2 }, Coord { x: 27.8, y: 0.1 }])
After: LineString([Coord { x: 0.0, y: 0.0 }, Coord { x: 5.0, y: 4.0 }, Coord { x: 11.0, y: 5.5 }, Coord { x: 17.3, y: 3.2 }, Coord { x: 27.8, y: 0.1 }])
```

This is probably related to #943 - I haven't verified it yet, but that's the only thing between the two versions that changed simplify code that I've noticed.

### PR DESCRIPTION
**Pull request by michaelkirk**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/main/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

FIXES #995

/cc @rjmac

### CODE REVIEW

File: geo/src/algorithm/simplify.rs [hunk 1]
```diff
@@ -293,13 +304,14 @@ mod test {
...
-        let vec = [
+        let input = [
             coord! { x: 8.0, y: 100.0 },
             coord! { x: 9.0, y: 100.0 },
             coord! { x: 12.0, y: 100.0 },
         ];
-        let simplified = rdp::<_, _, 2>(vec.into_iter(), &1.0);
-        assert_eq!(simplified, vec);
+        let actual = rdp::<_, _, 2>(input.into_iter(), &1.0);
+        let expected = [coord! { x: 8.0, y: 100.0 }, coord! { x: 12.0, y: 100.0 }];
```
Comment (michaelkirk) [inline geo/src/algorithm/simplify.rs:313]: This test was originally added in c61abfa1e758675cc6aa9925e3e6468cf104eaa4, and changed (I think incorrectly) in 9d81bf49ed31fdfeb76728563c8236879c2e2473 as part of #943

This reverts it to it's previous behavior.

**Review by urschrei — APPROVED** [review]

**Comment by urschrei** [comment]
Good catch @michaelkirk!

**Comment by michaelkirk** [comment]
Did you want to take a look at this too @frewsxcv?

**Review by frewsxcv — APPROVED** [review]

**Comment by frewsxcv** [comment]
Thoughts on pushing out a new bugfix release?

**Pull request merged**